### PR TITLE
 Refactor card components and optimize app display layout for app page. 

### PR DESCRIPTION
**Pull request by daniel-farina**
The following PR addresses some issues with the thumbnails views on the apps page on multiple displays but focused mostly on sm and lg devices. 

Here is a summary of the changes made in the affected files:

**packages/web/components/cards/app-display-card.tsx:**
Updated the CSS styles for the card image, including background size, position, and repeat.
Adjusted the height of the card image container.
Modified the minimum height of the card's content container.

**packages/web/components/cards/hero-card.tsx:**
Modified the CSS styles for the hero image container, including background size, position, and repeat.
Adjusted the height of the hero image container for different screen sizes.
Added a gradient overlay to the hero image.
Adjusted the layout and spacing of the card's content.

**packages/web/pages/apps/index.tsx:**
Updated the grid layout for displaying app cards.
Adjusted the number of columns and the gap between the cards.
Modified the container and column widths for different screen sizes.
Updated the styling and layout of the "getFeatured" section.
These changes aim to enhance the visual presentation and overall user experience of the app display cards in the web components. The refactored code has been tested and verified to maintain the expected functionality.

Full page
image

Smaller devices reduced to two columns to prevent odd sizing
image

Mobile
image

Please review and merge accordingly. 

TODO suggestions but not a blocker to merge:
- The mobile hero images for Mars needs all the text deleted, it's redundant. 
- The TFM thumbnails has the wrong proportions (it's a square), needs to be updated @JeremyParish69

### CODE REVIEW

**Review by jonator — DISMISSED** [review]
LGTM

But I think in the figma only the top corners of images are rounded:
Screenshot 2023-05-31 at 6 33 38 AM

**Comment by daniel-farina** [comment]
> LGTM
> 
> But I think in the figma only the top corners of images are rounded: Screenshot 2023-05-31 at 6 33 38 AM

Rounded corners are fine according to an updated Figma from Syed 👍

**Review by arispech — APPROVED** [review]
LGTM

**Review by MaxMillington — APPROVED** [review]
LGTM but let's get Syed to confirm. There are all sorts of things we can do with the images, but ultiimately there is only so much we can do since we have different aspect ratios

File: packages/web/components/cards/hero-card.tsx [hunk 1]
```diff
@@ -45,13 +45,15 @@ export const HeroCard: React.FunctionComponent<{
...
         onClick={handleAppClicked}
-        className="heroImage relative flex h-[400px] cursor-pointer items-end overflow-hidden rounded-lg"
+        className="heroImage relative flex h-[400px]  cursor-pointer items-end overflow-hidden rounded-lg sm:h-[300px]"
       >
-        <img
-          className="backgroundImage absolute top-0 left-0 z-10 h-full w-full bg-cover bg-center bg-no-repeat"
-          src={imageUrl}
-          alt="hero image"
-        />
+        <div
```
Comment (MaxMillington) [inline packages/web/components/cards/hero-card.tsx:50]: why change `img` to `div`?

Comment (daniel-farina) [reply packages/web/components/cards/hero-card.tsx:50]: The image behaved strangengly when it comes to automatic resizing via Tailwind and we had it twice (background and actual image ) which caused a double image effect for thumbnails that had transparency on them like the Kado one.

Comment (MaxMillington) [reply packages/web/components/cards/hero-card.tsx:50]: 👍

File: packages/web/components/cards/app-display-card.tsx [hunk 2]
```diff
@@ -41,18 +41,17 @@ export const AppDisplayCard: FunctionComponent<{
...
-        <div className="app-display-card bg-white h-[280px] overflow-hidden rounded-2xl bg-osmoverse-800 shadow-md lg:h-[300px]">
-          <div className="overflow-hidden">
-            <img
-              className="card-image inset-0 h-40 w-full overflow-hidden rounded-2xl bg-cover bg-center transition-transform duration-300 ease-in"
-              src={imageUrl}
-              alt="card image"
-            />
+        <div className="app-display-card bg-white min-h-[320px] overflow-hidden rounded-2xl bg-osmoverse-800 shadow-md xl:min-h-[320px] lg:min-h-[320px] md:min-h-[360px] sm:min-h-[290px] xs:min-h-[330px]">
+          <div className="overflow-hidden rounded-2xl">
+            <div className="card-image  min-h-[190px] xl:min-h-[180px] lg:min-h-[140px] md:min-h-[210px]  sm:min-h-[160px]  xs:min-h-[210px]"></div>
```
Comment (MaxMillington) [inline packages/web/components/cards/app-display-card.tsx:52]: @daniel-farina I didn't think of this before, but this is actually going to result in the heights of the cards being different.

Comment (MaxMillington) [reply packages/web/components/cards/app-display-card.tsx:52]: Meant to put this comment on the `app-display-card` <div>

**Pull request merged**